### ISSUE TITLE
Asset alt text missing from REST API response when querying an Entry

### ISSUE THREAD

**Issue by joshuablum** [statamic/cms#10011]
### Bug description

When querying the `/api/collections/{collection}/entries` endpoint and the entries include an asset, the alt text is missing from the response. When querying the asset directly (e.g. `/api/assets/{container}/{path}`) it's there.

Querying the collection and its entries (or just that specific entry by itself):
```json
{
  "assets_field":
    [
      {
        "id":"assets::test.png",
        "some more stuff but no alt text"
      }
    ]
}
```

Querying the asset works as expected:
```json
{
  "alt":"Wow, this is an Alt text.",
  "id":"assets::test.png"
}
```

### How to reproduce

1. Query the `/api/collections/{collection}/entries` endpoint with the blueprint having an asset field
2. Have an alt text for the image
3. See that the response does not include the alt text

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 10.48.10
PHP Version: 8.3.6
Composer Version: 2.7.1
Environment: local
Debug Mode: ENABLED
URL: sandbox.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 0
Antlers: runtime
Sites: 1
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.57.3 PRO
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None
